tests/PersonDialogTest.java

```diff
@@ -0,0 +1,204 @@
+import org.assertj.swing.core.matcher.JButtonMatcher;
+import org.assertj.swing.edt.FailOnThreadViolationRepaintManager;
+import org.assertj.swing.edt.GuiActionRunner;
+import org.assertj.swing.fixture.DialogFixture;
+import org.assertj.swing.fixture.FrameFixture;
+import org.assertj.swing.security.NoExitSecurityManagerInstaller;
+import org.junit.Rule;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static java.awt.event.KeyEvent.*;
+
+
+import java.io.IOException;
+
+class PersonDialogTest {
+
+    @Rule
+    private static FrameFixture window = null;
+    public static TemporaryFolder folder = new TemporaryFolder();
+
+
+    @BeforeAll
+    public static void init() {
+        // Prevent program exiting
+        NoExitSecurityManagerInstaller.installNoExitSecurityManager();
```

Comment:
Not all that important, but I figure I'd point out that I used this because there were lambda functions for exiting the program which needed checking, and That couldn't be done without this line